### PR TITLE
feat(prettier): Doc::IfBreak should a accept single doc and add if_break macro

### PR DESCRIPTION
**Pull request by Dunqing**

### CODE REVIEW

**Comment by Dunqing** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1335** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/1335?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #1336** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/1336?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/oxc-project/oxc/1335?utm_source=stack-comment).

File: crates/oxc_prettier/src/doc.rs [hunk 1]
```diff
@@ -31,15 +31,10 @@ pub enum Doc<'a> {
     /// no matter if the expression fits on one line or not.
     Hardline,
     /// Print something if the current `group` or the current element of `fill` breaks and something else if it doesn't.
-    IfBreak(Vec<'a, Doc<'a>>),
+    IfBreak(Box<Doc<'a>>),
```
Comment (Boshen) [inline crates/oxc_prettier/src/doc.rs:34]: This should use `Box` from the allocator.

The whole IR should be allocated on the allocator. I'll fix it after your merge.

**Comment by Boshen** [comment]
Nice catch, the doc change is correct, we need to look at the jsdoc lol

```
/**
 * @param {Doc} breakContents
 * @param {Doc} [flatContents]
 * @param {object} [opts] - TBD ???
 * @returns Doc
 */
function ifBreak(breakContents, flatContents = "", opts = {}) {
  assertDoc(breakContents);
  if (flatContents !== "") {
    assertDoc(flatContents);
  }

  return {
    type: DOC_TYPE_IF_BREAK,
    breakContents,
    flatContents,
    groupId: opts.groupId,
  };
}
```

**Comment by Boshen** [comment]
### Merge activity

* **Nov 15, 10:03 PM**: @Boshen started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/oxc-project/oxc/1335).

**Pull request merged**